src/backend/commands/createas.c

```diff
@@ -1159,6 +1161,38 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *ctx, int
 								 errmsg("expression containing an aggregate in it is not supported on incrementally maintainable materialized view")));
 					check_ivm_restriction_walker((Node *) tle->expr, ctx, depth);
 				}
+				/* additional restriction checks for exists subquery */
+				if (ctx->exists_quals != NIL && depth == 0)
+				{
+					ListCell *lc;
+
+					foreach (lc, ctx->exists_quals)
+					{
+						Var	*var = (Var *) lfirst(lc);
+						ListCell *lc2;
+						bool found = false;
+
+						foreach(lc2, qry->targetList)
+						{
+							TargetEntry	*tle = lfirst(lc2);
+							Var *var2;
+
+							if (!IsA(tle->expr, Var))
+								continue;
+							var2 = (Var *) tle->expr;
+							if (var->varno == var2->varno && var->varattno == var2->varattno)
+							{
+								found = true;
+								break;
+							}
+						}
+						if (!found)
+							ereport(ERROR,
+									(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+									 errmsg("this query is not allowed on incrementally maintainable materialized view"),
+									 errhint("targetlist must contain vars what is reffered by subquery")));
```

Comment:
targetlist must contain vars that are referred to in EXISTS subquery

Reply:
Thank you for recommendation. I will fix it.